base/compiler/ssair/inlining.jl

```diff
@@ -1182,25 +1189,32 @@ function analyze_single_call!(ir::IRCode, todo::Vector{Pair{Int, Any}}, idx::Int
     return nothing
 end
 
-function handle_const_call!(ir::IRCode, idx::Int, stmt::Expr,
+function maybe_handle_const_call!(ir::IRCode, idx::Int, stmt::Expr,
         info::ConstCallInfo, sig::Signature, @nospecialize(calltype),
         et::Union{EdgeTracker, Nothing}, caches::Union{InferenceCaches, Nothing},
         isinvoke::Bool, todo::Vector{Pair{Int, Any}})
-    item = InliningTodo(info.result, sig.atypes, calltype)
-    validate_sparams(item.mi.sparam_vals) || return
+    # TODO effectively use mutliple constant analysis results
+    length(info.results) == 1 || return false
+    result = info.results[1]
+    isa(result, InferenceResult) || return false
```

Comment:
@Keno do you have any idea of how we can effectively use multiple constant analysis results ?